astroquery/desi/core.py

```diff
@@ -0,0 +1,140 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+
+import os
+import shutil
+import tempfile
+import urllib.error
+
+from astropy.io.fits import HDUList
+import requests
+from astropy.table import Table, vstack
+
+from astroquery.exceptions import NoResultsWarning
+from ..query import BaseQuery
+from ..utils import commons
+from ..utils import async_to_sync
+from . import conf
+
+import pyvo as vo
+import numpy as np
+
+__all__ = ['DESILegacySurvey', 'DESILegacySurveyClass']
+
+
+@async_to_sync
+class DESILegacySurveyClass(BaseQuery):
+    URL = conf.server
+    TIMEOUT = conf.timeout
+
+    def query_region_async(self, coordinates, radius, data_release=9):
+        """
+        Queries a region around the specified coordinates.
+
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates`.
+            coordinates around which to query
+        radius : str or `astropy.units.Quantity`.
+            the radius of the cone search
+
+        Returns
+        -------
+        response : `requests.Response`
+            The HTTP response returned from the service.
+            All async methods should return the raw HTTP response.
```

Comment:
This function does not seem to return an HTTP response as it should if it is going to be an async function.